Tests/CommandsTests/PackageToolTests.swift

```diff
@@ -489,6 +490,37 @@ final class PackageToolTests: XCTestCase {
         }
     }
 
+    func testWatchmanXcodeprojgen() {
+        mktmpdir { path in
+            let fs = localFileSystem
+            let diagnostics = DiagnosticsEngine()
+
+            let scriptsDir = path.appending(component: "scripts")
+            let packageRoot = path.appending(component: "root")
+
+            let helper = WatchmanHelper(
+                diagnostics: diagnostics,
+                watchmanScriptsDir: scriptsDir,
+                packageRoot: packageRoot)
+
+            let script = try helper.createXcodegenScript(
+                XcodeprojOptions())
+
+            XCTAssertEqual(try fs.readFileContents(script), """
+                #!/usr/bin/env bash
+
+
+                # Autogenerated by SwiftPM. Do not edit!
+
+
+                set -eu
+
+                swift package generate-xcodeproj
```

Comment:
Do we also want a test that makes such that we pass `--xcconfig-overrides` when necessary?